engine-tests/src/tests/transaction.rs

```diff
@@ -161,6 +173,129 @@ fn test_access_list_tx_encoding_decoding() {
+fn test_eip_7702_tx_encoding_decoding() {
+    let secret_key = example_signer().secret_key;
+    let transaction = eip7702_transaction(0);
+
+    let signed_tx = utils::sign_eip_7702_transaction(transaction, &secret_key);
+    let tx_bytes: Vec<u8> = iter::once(eip_7702::TYPE_BYTE)
+        .chain(rlp::encode(&signed_tx))
+        .collect();
+
+    let decoded_tx = match EthTransactionKind::try_from(tx_bytes.as_slice()) {
+        Ok(EthTransactionKind::Eip7702(tx)) => tx,
+        Ok(_) => panic!("Unexpected transaction type"),
+        Err(e) => panic!("Transaction parsing failed: {e:?}"),
+    };
+
+    assert_eq!(signed_tx, decoded_tx);
+    assert_eq!(
+        signed_tx.sender().unwrap(),
+        utils::address_from_secret_key(&secret_key)
+    );
+}
+
+#[test]
+fn test_eip_7702_success() {
+    // 0xa52a8a2229e3c512d6ed27b6e6e7d39958ca9fb3,
+    let mut runner = utils::deploy_runner();
+    let signer = example_signer();
+    let signer_address = utils::address_from_secret_key(&signer.secret_key);
+    let contract_address = utils::address_from_hex(CONTRACT_ADDRESS);
+    let contract_code = hex::decode(CONTRACT_CODE_EIP7702).unwrap();
+
+    runner.create_address(signer_address, INITIAL_BALANCE, signer.nonce.into());
+    runner.create_address_with_code(
+        contract_address,
+        CONTRACT_BALANCE,
+        CONTRACT_NONCE.into(),
+        contract_code.clone(),
+    );
+
+    let mut transaction = eip7702_transaction(0);
+    transaction.chain_id = runner.chain_id;
+    let signed_tx = utils::sign_eip_7702_transaction(transaction, &signer.secret_key);
+    let tx_bytes: Vec<u8> = iter::once(eip_7702::TYPE_BYTE)
+        .chain(rlp::encode(&signed_tx))
+        .collect();
+
+    let sender = "relay.aurora";
+    let outcome = runner.call(utils::SUBMIT, sender, tx_bytes).unwrap();
+    // Unwrapping execution results validates outcome
+    let result = SubmitResult::try_from_slice(&outcome.return_data.as_value().unwrap()).unwrap();
+
+    let delegated_designator = Address::decode("d16292912e440956828f1a847ca6efc8412f45b6").unwrap();
+
+    assert_eq!(result.gas_used, 68206);
+    assert_eq!(runner.get_nonce(signer_address), (signer.nonce + 1).into());
+    assert_eq!(runner.get_balance(contract_address), CONTRACT_BALANCE);
+    assert_eq!(runner.get_nonce(contract_address), CONTRACT_NONCE.into());
+    assert_eq!(runner.get_code(contract_address), contract_code);
+    // `EXTCODESIZE` should return size of `EF01` = 2 for delegated designator
+    assert_eq!(
+        runner.get_storage(contract_address, H256::zero()),
+        H256::from(H160::from_low_u64_be(2))
+    );
+
+    // Authority address should increase Nonce
+    assert_eq!(runner.get_nonce(delegated_designator), 1.into());
+    // Get delegated designator address
+    assert_eq!(
+        hex::encode(runner.get_code(delegated_designator)),
+        "ef0100cccccccccccccccccccccccccccccccccccccccc"
+    );
+}
+
+#[test]
+fn test_eip_7702_wrong_auth_nonce() {
```

Comment:
Is the name of this test correct? It looks to me like the reason the authorization fails is because the chain id is wrong not because the nonce is wrong.

Reply:
Absolutely! Both changes in: 69dd68a764d13ae1b2b4efc2868e21d5be602004